### PR TITLE
fix: non-semantic version automatic deployments do not work when version labels are not valid semantic versions

### PR DESCRIPTION
**Pull request by sgalsaleh**
#### What this PR does / why we need it:

Enabling automatic deployments when semantic versioning is not enabled does not automatically deploy new versions if the version label is not a valid semver. This PR will allow the automatic deployment to go through even if the version label is not a valid semantic version.

This PR also addresses another issue where any new version would be automatically deployed even if it's not a new upstream release (e.g. a config change, license change, etc...).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fixes an issue where non-semantic version automatic deployments do not work when version labels are not valid [SemVer](https://semver.org/).
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/updatechecker/updatechecker.go [hunk 1]
```diff
@@ -472,21 +472,23 @@ func autoDeploy(opts CheckForUpdatesOpts, clusterID string, autoDeploy apptypes.
...
 
-	// In the case semver is not required, we only need to check if the newest app version
-	// is greater than the current version.
 	if autoDeploy == apptypes.AutoDeploySequence {
-		if currentVersion.Sequence < appVersions.AllVersions[0].Sequence {
+		// semver is not required/enabled, we only need to check if the newest app version is newer than the current version.
+		// use cursor instead of sequence in order to only deploy newer upstream versions, and not versions created by config changes, license changes, etc...
+		currentCursor := currentVersion.Cursor
+		latestCursor := appVersions.AllVersions[0].Cursor
+		if currentCursor != nil && latestCursor != nil && (*currentCursor).Before(*latestCursor) {
```
Comment (divolgin) [inline pkg/updatechecker/updatechecker.go:486]: Should this be determined by that weird sorting algorithm we have?

Comment (sgalsaleh) [reply pkg/updatechecker/updatechecker.go:486]: that algorithm will sort using sequences and not cursors, which is what causes the issue of automatically deploying config changes, license changes, etc...

this code block is only relevant for when semantic versioning is disabled any ways

Comment (sgalsaleh) [reply pkg/updatechecker/updatechecker.go:486]: also, `appVersions.AllVersions` is already sorted using that algorithm

**Review by divolgin — DISMISSED** [review]

File: pkg/updatechecker/updatechecker.go (hunk 1 above)
Comment (sgalsaleh) [reply pkg/updatechecker/updatechecker.go:486]: also, `appVersions.AllVersions[0]` is the top version after sorting, so it for sure is higher than or equal to the current version in terms of order. then the only question becomes whether or not it has a higher cursor if it's a variation of the same upstream version.

**Comment by dexhorthy** [comment]
I'm Dexter Horthy and I approve this PR

**Review by divolgin — APPROVED** [review]

**Pull request merged**